### DISCUSSION TITLE
Implement issue template

### DISCUSSION THREAD

**Discussion by StefanVDWeide** [jakeboone02/parse-ingredient#6] [Ideas]
Hi! 

I've been looking for a good ingredient parsing library for a while and found that a lot of them either aren't maintained anymore or have features fragmented over different forks with no intention to merge back. This project has good potential to become the standard ingredient parsing library! 

Now, if this library gets more populair it would be a good idea to have an issue template so that everything is nice and clean. 

Details on how to implement this can be found here: 

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository

Let me know what you think!

**Comment by jakeboone02** [comment]
Thanks for your interest in the library! With a grand total of 30 downloads on npm in the last 7 days, 8 stars on GitHub, and #5 being the very first issue raised after its initial release more than two years ago...I'm not really sure that I would benefit a great deal from an issue template.

I do appreciate your enthusiasm, though! I'm quite proud of this library and its companion packages `numeric-quantity` and `format-quantity`, even considering their low engagement numbers.

**Reply by StefanVDWeide** [reply]
Fair enough, maybe it's a bit too much for now. But who knows ;) 

I was especially drawn to this library over the alternatives because of the demo page which immediately lets me verify if it outputs what I want with the recipes I'm working with. Also, like I mentioned, the alternatives have quite some forks with different features and no real clear winner. So, you are right to be proud! 

I'm happy to open PRs and improve it further as I implemented it into my project if you are okay with keeping an eye on the library and reviewing + merging them.

**Reply by jakeboone02** [reply]
Yes, as I said feel free to submit PR's and any other comments you have. I'm glad you're liking it!

As an aside, are there any missing features you'd like to see added? Anything we can "steal" from the alternatives?

**Reply by StefanVDWeide** [reply]
So [this library](https://github.com/suprmat95/recipe-parser) is an interesting one to "steal from". It supports:

- Localization 
- Combining ingredient objects 

I've also noticed some recipes saying something like `2 large or 3 medium leeks, white and pale green parts only, halved lengthwise, thinly sliced` which doesn't seem to be supported by anyone yet. Maybe worth looking at, I'll open a dedicated issue for this.

**Reply by jakeboone02** [reply]
That's an interesting one with the natural language processing, but I'm not sure NLP is necessary for such a narrow, (typically) structured case as ingredient lists.

I _think_ we can already do some limited l10n/i18n with the `additionalUOMs` parameter. It wouldn't handle leading "of" (e.g. "di"/"d'" in Italian), but the units of measure would be taken care of.

The one thing I'd really like to "steal" is support for spelled-out quantities like "six cups flour". That would require mapping all the small numbers and magnitude numbers (and even more for certain languages) from a language, so I'd rather add that as another configuration option instead of hard-coding each supported language. We'd default to English of course, but allow additional languages based on some specific format. Maybe we could provide configurations of some of the more common languages like Italian, French, etc.

Thoughts?

**Reply by StefanVDWeide** [reply]
Agree on the NLP part, like you said the structure that already exists should be good enough for us. The only exceptions are like the example I gave where the recipe author says something like `2 large or 3 medium leeks`. Although looking for `small`, `medium` and `large` during the processing should cover this pretty well.

As for i10n/i18n, also agree here. The alternates should cover translated UOM names and having a config option that defaults to English for written out numbers sounds like a solid place to start. Having both of these behind config options would also be my approach.

A different question, is there an automated release process in place for generating a new release and publishing to NPM after a PR is  merged into the main branch and passes the tests?

**Reply by jakeboone02** [reply]
I just added "small", "medium", and "large" as units of measure.

No, there's no automated release process.

**Reply by jakeboone02** [reply]
FYI: [v0.6.0](https://github.com/jakeboone02/parse-ingredient/releases/tag/v0.6.0)

**Reply by dRoskar** [reply]
Hello to both.
I've also taken interest in this library. The reason I am reaching out here is because of the change you mentioned in this thread. Adding "small", "medium" and "large" to the units list breaks this library for me. When the ingredients list says "2 large eggs" the unit of measure should be null, in my opinion. I don't quite understand how adjectives became units of measure.
But in case this is needed for some use case that I'm not aware of, at least it would be great to give users the option to exclude certain units of measure from the list, just like we can add them.
Best regards

**Reply by jakeboone02** [reply]
@dRoskar That's a good point about adjectives not being units of measure. I think it works in this case because it can imply a particular amount of something, not just a relative size. For example, chicken eggs have to be within [very specific mass limits](https://en.wikipedia.org/wiki/Chicken_egg_sizes) to be labeled as "small"/"medium"/"large"/etc.

But I also think an "escape hatch" would be a nice feature. Maybe an array of strings to ignore when looking for units? Something like this:

```ts
parseIngredient('2 large eggs', { ignoreUOM: [ 'small', 'medium', 'large' ] })
```

What do you think?

**Reply by dRoskar** [reply]
I think that would work perfectly 👍

**Reply by jakeboone02** [reply]
See #17
